### DISCUSSION TITLE
Zen subwindow rules

### DISCUSSION THREAD

**Discussion by prochy-exe** [hyprwm/Hyprland#8078] [Q&A]
Heya! I'm trying to make a couple of rules for the Zen browser, based on Firefox, but I can't seem to get them working. 

I am using the git version of hyprland.

I am trying to match this window and make it floating

```
{
  "address": "0x607f454e3d20",
  "mapped": true,
  "hidden": false,
  "at": [
    1927,
    7
  ],
  "size": [
    946,
    1066
  ],
  "workspace": {
    "id": 2,
    "name": "2"
  },
  "floating": false,
  "pseudo": false,
  "monitor": 1,
  "class": "zen-alpha",
  "title": "Page Info — https://www.reddit.com/r/hyprland/comments/1fvxemm/help_how_to_find_title_for_an_app_for_window_rules/",
  "initialClass": "zen-alpha",
  "initialTitle": "",
  "pid": 15633,
  "xwayland": false,
  "pinned": false,
  "fullscreen": 0,
  "fullscreenClient": 0,
  "grouped": [],
  "tags": [],
  "swallowing": "0x0",
  "focusHistoryID": 0
}
```

and the rule I've written is this `windowrulev2 = float, title:^(Page Info)(.*)$, class:^(zen-alpha)$`

The strange thing is that when i do a rule with just `windowrulev2 = float,class:^(zen-alpha)$` the rule gets applied just fine, but obviously for all Zen browser windows.

Am I missing something? I checked if the regex matches using regex101 and it matched just fine so I'm sort of lost.

Thanks!

**Comment by fufexan** [comment]
I have the same problem. The issue is the initial title being empty. We have to make a script that listens to socket2 and applies the rule after the title has been set.

**Reply by prochy-exe** [reply]
Well that's odd, because I tried `windowrulev2 = float, title:^(Picture-in-Picture)(.*)$, class:^(zen-alpha)$` and it works properly, but it does have the intialTitle set. 

```
{
  "address": "0x5a4124591450",
  "mapped": true,
  "hidden": false,
  "at": [
    967,
    547
  ],
  "size": [
    946,
    526
  ],
  "workspace": {
    "id": 1,
    "name": "1"
  },
  "floating": false,
  "pseudo": false,
  "monitor": 0,
  "class": "zen-alpha",
  "title": "Picture-in-Picture",
  "initialClass": "zen-alpha",
  "initialTitle": "Picture-in-Picture",
  "pid": 863,
  "xwayland": false,
  "pinned": false,
  "fullscreen": 0,
  "fullscreenClient": 0,
  "grouped": [],
  "tags": [],
  "swallowing": "0x0",
  "focusHistoryID": 1
}
```

I tried to write the same rule for a bitwarden popup window but that doesn't work
```
{
  "address": "0x5a41244dd340",
  "mapped": true,
  "hidden": false,
  "at": [
    2887,
    7
  ],
  "size": [
    946,
    1066
  ],
  "workspace": {
    "id": 2,
    "name": "2"
  },
  "floating": false,
  "pseudo": false,
  "monitor": 1,
  "class": "zen-alpha",
  "title": "Extension: (Bitwarden Password Manager) - Bitwarden — Zen Browser",
  "initialClass": "zen-alpha",
  "initialTitle": "Zen Browser",
  "pid": 863,
  "xwayland": false,
  "pinned": false,
  "fullscreen": 0,
  "fullscreenClient": 0,
  "grouped": [],
  "tags": [],
  "swallowing": "0x0",
  "focusHistoryID": 3
}
```

the rule i tried `windowrulev2 = float, title:^(Extension: \(Bitwarden Password Manager\) - Bitwarden — Zen Browser)(.*)$, class:^(zen-alpha)$`

very odd, I don't see any reason why it shouldn't work, maybe a firefox bug because I also tried the same thing with Firefox and it does the same thing

**Reply by biohackerellie** [reply]
not sure if you figured it out yet but found your question looking for the same thing, and i ended up getting it working like this `windowrulev2 = float,class:^(zen-beta)$,title:^(Extension: \(Bitwarden Password Manager\))$
`

**Reply by fufexan** [reply]
Is `initialTitle` being set properly for you then?

**Reply by biohackerellie** [reply]
seems so yea

**Reply by fufexan** [reply]
Interesting. I'll have to check why the zen from the flake does not (and whether firefox has initialTitle set).

**Reply by Mondotosz** [reply]
I'm using zen on arch (the zen-browser-bin package) and the `initialTitle` is also not set properly on both zen and firefox

**Reply by RoyvanEmpel** [reply]
Wow this still seems to be a problem in 2026! I'm guessing that this is only fixable by ZEN it self so i should maybe open a ticket there (if there isn't already). Or is this a general firefox issue that needs to be fixed by them since both Zen and Firefox have the same issue?

> 	class: app.zen_browser.zen
	title: Extension: (Bitwarden Password Manager) - Bitwarden — Zen Browser
	initialClass: app.zen_browser.zen
	initialTitle: Zen Browser

> 	class: firefox
	title: Extension: (Bitwarden Password Manager) - Bitwarden — Mozilla Firefox
	initialClass: firefox
	initialTitle: Mozilla Firefox

Als a workaround I did find this solution on an github issue, it isn't pretty but it works.
https://github.com/hyprwm/Hyprland/issues/3835#issuecomment-2004448245

**Reply by RoyvanEmpel** [reply]
Oh, modern version for lua: https://github.com/hyprwm/Hyprland/issues/3835#issuecomment-4435390873
